File: gmlrva-sample/src/main/java/pt/simdea/gmlrva/sample/data/FakeDataObject.java
/*
 * Copyright (c) 2017. Simdea.
 */

package pt.simdea.gmlrva.sample.data;

import android.support.annotation.NonNull;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;

/**
 * Class meant to model a random fake data object.
 *
 * Created by Paulo Ribeiro on 9/26/2017.
 * Simdea © All Rights Reserved.
 * paulo.ribeiro@simdea.pt
 */
@AllArgsConstructor
@Getter
@EqualsAndHashCode(exclude = "mDescription")
public final class FakeDataObject implements Comparable<FakeDataObject> {

    private String mTitle;
    private String mDescription;

    @Override
    public int compareTo(@NonNull final FakeDataObject o) {
        return getTitle().compareToIgnoreCase(o.getTitle());
    }

}

File: gmlrva-sample/src/main/java/pt/simdea/gmlrva/sample/data/FakeDataObject.kt
/*
 * Copyright (c) 2017. Simdea.
 */

package pt.simdea.gmlrva.sample.data

/**
 * Class meant to model a random fake data object.
 *
 * Created by Paulo Ribeiro on 9/26/2017.
 * Simdea © All Rights Reserved.
 * paulo.ribeiro@simdea.pt
 */
class FakeDataObject(private val mTitle: String, private val mDescription: String) : Comparable<FakeDataObject> {

    override fun compareTo(other: FakeDataObject): Int {
        return mTitle.compareTo(other.mTitle, true)
    }

}
